src/net.h

```diff
@@ -1396,6 +1396,13 @@ class CConnman
     // P2P timeout in seconds
     std::chrono::seconds m_peer_connect_timeout;
 
+    /**
+     * Reachable networks that have addresses in the addrman (new, tried). It is filled when there
+     * are few reachable addresses in addrman and used to filter the addrman's selection.
+     * Otherwise, we avoid the cost of the filtering.
+     */
+    std::pair<std::unordered_set<Network>, std::unordered_set<Network>>  m_networks_with_addrs;
```

Comment:
nit: IMO the code will be more self-documenting if you use 2 separate variables for new vs tried, rather than a pair here, because to make sense of the pair you have to refer back to the definition, while new vs tried in the variable name will be immediately apparent.

Could still return a pair from `NetworksWithAddresses`, and unpack it for assignment.

Reply:
Sounds good, I'll address it.